great_expectations/execution_engine/redshift_execution_engine.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+import re
+from typing import Any, Optional, cast
+
+from great_expectations.compatibility import aws
+from great_expectations.compatibility.sqlalchemy import sqlalchemy as sa
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.exceptions.exceptions import GreatExpectationsError
+from great_expectations.execution_engine.sqlalchemy_batch_data import SqlAlchemyBatchData
+from great_expectations.execution_engine.sqlalchemy_execution_engine import (
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.metric_provider import metric_value
+from great_expectations.expectations.metrics.table_metrics.table_column_types import (
+    ColumnTypes as BaseColumnTypes,
+)
+
+
+class RedshiftExecutionEngine(SqlAlchemyExecutionEngine):
+    """SqlAlchemyExecutionEngine for Redshift databases."""
+
+    pass
+
+
+# The complete list of reshift types can be found here:
+# https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html
+REDSHIFT_TYPES = (
+    {
+        "boolean": aws.redshiftdialect.BOOLEAN,
+        "smallint": aws.redshiftdialect.SMALLINT,
+        "integer": aws.redshiftdialect.INTEGER,
+        "bigint": aws.redshiftdialect.BIGINT,
+        "real": aws.redshiftdialect.REAL,
+        "double precision": aws.redshiftdialect.DOUBLE_PRECISION,
+        # Numeric is odd since the data returns numeric as a type to pg_get_cols
+        # In sqlalchemy 1.4 we get the type sa.sql.sqltypes.NUMERIC returned.
+        # However, the sqlalchemy redshift dialect only has DECIMAL. The official
+        # redshift docs say 'DECIMAL' is the correct type and NUMER is an alias:
```

Comment:
This DECIMAL/NUMERIC inconsistency within the dialect is a little strange and we can fix up our fork once that is up if we want to follow up on this.